lib/Rex/Resource/file/Provider/base.pm

```diff
@@ -76,7 +76,10 @@ sub test {
   my $is_dir    = $self->is_dir;
   my $is_absent = any { $self->config->{ensure} eq $_ } qw/absent/;
 
-  if ( $is_file && $self->fs->is_file($file) ) {
+  if ( $is_file
+    && $self->fs->is_file($file)
+    && ( $self->config->{source} || $self->config->{content} ) )
```

Comment:
Looks like the new extra condition also skips file stat comparisons, not just checksum comparison: https://github.com/slashbeast/Rex/blob/f09a95d8b0cced1c4a662464caa0d0398a304f9d/lib/Rex/Resource/file/Provider/base.pm#L82-L102

Can you make sure it only skips checksum calculation, please?

If you can, please add a test for this case first (no `content` or `source`, but at least one of `owner`/`group`/`mode` should be changed), but let me know if you would like me to add it.